modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/PythonClientExperimentalCodegen.java

```diff
@@ -829,13 +827,13 @@ public CodegenModel fromModel(String name, Schema schema) {
         result.unescapedDescription = simpleModelName(name);
 
         // make non-object type models have one property so we can use it to store enums and validations
-        if (result.isAlias || result.isEnum) {
+        if (result.isAlias || result.isEnum || result.isArrayModel) {
             Schema modelSchema = ModelUtils.getSchema(this.openAPI, result.name);
             CodegenProperty modelProperty = fromProperty("value", modelSchema);
-            if (modelProperty.isEnum == true || modelProperty.hasValidation == true) {
+            if (modelProperty.isEnum == true || modelProperty.hasValidation == true || result.isArrayModel) {
                 // these models are non-object models with enums and/or validations
                 // add a single property to the model so we can have a way to access validations
-                result.isAlias = true;
```

Comment:
Array models are making a model that aliases the array type, so I think that this should be true.
Long term my goal is to have the ModelSimple classes inherrit from their base classes like str int etc.
Why do we need to set this to false for Arraymodels? Historically generateAliasAsModel applied only to Map and Array models in most generators. In python-experimental we apply it to those and all primitives only if they have validations and ebums. Standard processing in the generators was to describe all these primitive models as their simple types like str, list. But if those models had validations and enums that data was lost in that process. So I added code to preserve that information and make object type models which preserve the validation information which would have been lost.

After I did that CodegenModel was updated to include validations which previously had been missing. So in summary this code here is an older hack to preserve information. If we can use the existing ArrayModels as is that is preferable. It should define its datatype somewhere inside the CodegebModel and that should be used to set the ModelSimple data type in the python class for our made up calue parameter.

We can do this by doing a check of the model type when we assign the openapi types in ModelSimple.
We can check the data that is available by running the generator with a debugmodels flag.

Reply:
I am not aware of all the corner cases in the code. I was trying to make it work for my use-case and not break existing tests. Feel free to propose code changes.

Reply:
I hope I have found a sufficient fix in b2ca2f78e295ce1d94b42f6ebc55d9d5a64f1918.